ai.py

```diff
@@ -247,9 +248,20 @@ async def on_ready():
         asyncio.ensure_future(storage.release_timed(bot))
 
 
+@bot.event
+async def on_command_error(context, error):
+    LOGGER.error(f"!!! Exception caught in {context.command} command !!!")
+    LOGGER.error(error)
+    with open("ai.log", "a") as exception_file:
+        traceback.print_exception(type(error), error, error.__traceback__, file=exception_file)
+    traceback.print_exception(type(error), error, error.__traceback__, file=sys.stderr)
+
+
 @bot.event
 async def on_error(event, *args, **kwargs):
-    LOGGER.error(
-        f'Exception encountered while handling message with content [{args[0].content}] in channel [{args[0].channel.name}]', exc_info=True)
+    LOGGER.error(f'!!! EXCEPTION CAUGHT IN {event} !!!')
+    with open("ai.log", "a") as exception_file:
+        traceback.print_exc(file=exception_file)
+    traceback.print_exc(file=sys.stderr)
```

Comment:
I would prefer this, because it's all handled with the logger:
```suggestion
    LOGGER.error(f'!!! EXCEPTION CAUGHT IN {event} !!!')
    LOGGER.error(sys.exc_info(), exc_info=True, stack_info=True)
```